prometheus/client_test.go

```diff
@@ -0,0 +1,51 @@
+package prometheus
+
+import (
+	"github.com/swift-sunshine/swscore/config"
+)
+
+func setupConfig() {
+	conf := config.NewConfig()
+	conf.PrometheusServiceUrl = "http://prometheus-istio-system.127.0.0.1.nip.io"
+	config.Set(conf)
+}
+
+// For now it's just a runnable manual test rather than an itest.
```

Comment:
I also thought on these kind of manual tests.
I wonder if there is a way to uncomment them and just enable them with some flag on the config or prompt i.e. 'Manual/Whatever'.
In that way perhaps it would save time if you want to use these tests.

I don't know which is the best practice in golang, in maven we used to have these kind of variables for those manual tests.

Reply:
I think in these tests we should just mock the prometheus server. A simple testHttpServer would do the job and return some metrics (which can be then asserted against).